s3-plugin-server/src/main/scala/com/gu/teamcity/ArtifactUploader.scala

```diff
@@ -31,13 +33,21 @@ class ArtifactUploader(config: S3ConfigManager, s3: S3) extends BuildServerAdapt
                   report("Not configured for uploading")
                   Continuation.BREAK
                 }
-            } recover {
+            } recoverWith {
+              case e: ResetException => {
+                runningBuild.addBuildMessage(new BuildMessage1(DefaultMessagesInfo.SOURCE_ID,
+                  DefaultMessagesInfo.MSG_BUILD_PROBLEM, Status.WARNING, new Date,
+                  s"Retrying artifact upload after error: ${e.getMessage}"))
```

Comment:
This is reporting back in the UI, rather than logging. Given that we've already established it's a reset at this point, I thought we'd probably gain little with more than the message.